### PR TITLE
chore(setup CircleCI): Enable continuous integration

### PR DESCRIPTION
**Pull request by Etomovich**
**Description**
<hr/>

This PR creates the configuration of Circle CI. It also adds configuration of Code Climate that checks code maintainability and test coverage.

**Type of Change**
<hr/>

- [x] chore(non-breaking change that makes continuous integration possible).

**Checklist**
- [x] Integration of CircleCI

**Related Stories**
[Continous Intergration with Circle CI](https://www.pivotaltracker.com/n/projects/2313280/stories/164046207)

### CODE REVIEW

**Comment by abulojoshua1** [comment]
@Etomovich, kindly add a label to this PR.

**Comment by Etomovich** [comment]
> @Etomovich, kindly add a label to this PR.

Thanks for that @abulojoshua1. A slight oversight.

**Review by BrianSerem — APPROVED** [review]

**Review by ElMonstro — APPROVED** [review]

**Review by MandelaK — APPROVED** [review]

**Review by RachelleMaina — APPROVED** [review]
LGTM

**Review by cob04 — APPROVED** [review]
Circle Ci checks run fine.
Stubborn dependency `"eslint": "^5.16.0"` removed.

File: .gitignore [hunk 1]
```diff
@@ -16,8 +16,10 @@
...
 .env.development.local
 .env.test.local
-.env.production.local
+.env.production.local 
 
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+package-lock.json
```
Comment (abulojoshua1) [inline .gitignore]: @Etomovich, kindly add an EOF here.

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Etomovich) [reply .gitignore]: Ok sorry for the delay

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**